### PR TITLE
Tiny Feature : Pass "locals" as the <include/> inner text

### PR DESCRIPTION
**Pull request by felippe-regazio**
This is a simple PR to allow pass the "locals" as the <include/> inner text instead an attribute. I think it can improve readability and organization. There is only one caveat: The "\n" chars will be stripped from the data, so for the case when a "\n" is needed as part of data, the "locals" attribute can still be used.

### CODE REVIEW

File: test/expected/inner-locals.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<h1>index</h1><h2>3</h2>
```
Comment (Scrum) [inline test/expected/inner-locals.html:1]: result should be 

```html
<h1>index</h1>

<h2>3</h2>
```

File: test/fixtures/inner-locals.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<h1>index</h1>
+
+<include src="./test/fixtures/includes/3.html">
+  {
+    "text": 3
+  }
+</include>
```
Comment (Scrum) [inline test/fixtures/inner-locals.html:7]: This test will fail with an error `Unexpected token s in JSON at position 3`
```html
<include src="./test/fixtures/includes/3.html">
  simple: fail
</include>
```

https://github.com/posthtml/posthtml-include/blob/99549766beebc1acbc567da0e7fdc4209e17d6db/lib/index.js#L28-L29

I think it's worth changing the validation logic, something like this

```js
if (locals) {
  try {
    locals = JSON.parse(locals);
  } catch {
    locals = false;
  }
}
```

You also forgot to add this to the main `test/test.js` file
```js
test('Inner locals', t => {
  return process(t, 'inner-locals')
})
```

File: lib/index.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ module.exports = (options = {}) => {
 
     tree.match({tag: 'include'}, node => {
       let src = node.attrs.src || false;
-      let locals = node.attrs.locals || false;
+      let locals = node.attrs.locals || (node.content ? node.content.join().replace(/'\n'/g, '') : false);
```
Comment (Scrum) [inline lib/index.js:20]: I think it should be replaced according to the changes in the comment to

```js
let locals = node.attrs.locals ?? node.content?.join().replace(/\n/g, '');
```
> ⚠️ Pay attention to the regular expression, there are optional single quotes

**Review by Scrum — CHANGES_REQUESTED** [review]

File: lib/index.js (hunk 3 above)
Comment (felippe-regazio) [reply lib/index.js:20]: nice, i fixed the regex. wouldn't we need an ESx compiler (babel?) to user this syntax? the "npm run test" gets broken with error:

```js
✖  25:47  Parsing error: Unexpected token ? 
```

File: test/expected/inner-locals.html (hunk 1 above)
Comment (felippe-regazio) [reply test/expected/inner-locals.html:1]: fixed

File: test/fixtures/inner-locals.html (hunk 2 above)
Comment (felippe-regazio) [reply test/fixtures/inner-locals.html:7]: Thats true, i fixed it. I also made a tiny change on the logic when trying to parse "locals" to fit this needings. All the tests are passing now.

File: lib/index.js (hunk 3 above)
Comment (Scrum) [reply lib/index.js]: > wouldn't we need an ESx compiler (babel?) to use this syntax?

Yes, which version of the node are you using? This package starts support from 10 and up

File: test/fixtures/inner-locals.html (hunk 2 above)
Comment (Scrum) [reply test/fixtures/inner-locals.html:7]: add this test more coverage
```html
<include src="./test/fixtures/includes/3.html">
  simple: fail
</include>
```

File: lib/index.js [hunk 4]
```diff
@@ -17,16 +17,27 @@ module.exports = (options = {}) => {
...
     tree.match({tag: 'include'}, node => {
       let src = node.attrs.src || false;
-      let locals = node.attrs.locals || false;
       let content;
       let subtree;
       let source;
+      let locals;
+
+      const localsString = node.attrs.locals || (node.content ? node.content.join().replace(/\n/g, '') : '');
+      const tryToParseLocals = localsString.trim() && localsString.trim().startsWith('{');
```
Comment (Scrum) [inline lib/index.js:26]: I'm not sure if there is a need for this, for example, the following entry is possible
```
simple }{ test
```

**Review by Scrum — CHANGES_REQUESTED** [review]

File: lib/index.js (hunk 3 above)
Comment (felippe-regazio) [reply lib/index.js]: v10.16.0

Comment (Scrum) [reply lib/index.js]: yeahh, unfortunately they are supported from 14 version

File: lib/index.js (hunk 4 above)
Comment (felippe-regazio) [reply lib/index.js:26]: yep, in this case something like, maybe(?)

```js
     let locals;
      
      try {
       // keep on a const or pass the expression directly to JSON.parse() ?
        const localsString = node.attrs.locals || (node.content ? node.content.join().replace(/\n/g, '') : '');
        
        locals = JSON.parse(localsString);
      } catch {
        locals = false;
      }

      if (src) {
        src = path.resolve(options.root, src);
        source = fs.readFileSync(src, options.encoding);

        if (locals) {
          const result = posthtml()
            .use(expressions({locals}))
            .process(source, {sync: true});
          source = result.html;
        }
```

Comment (felippe-regazio) [reply lib/index.js:26]: yep, in this case maybe something like:

```js
      // ...
      let locals;

      if (src) {
        src = path.resolve(options.root, src);
        source = fs.readFileSync(src, options.encoding);
        locals = node.attrs.locals || (node.content ? node.content.join().replace(/\n/g, '') : false);

        try {
          locals = JSON.parse(locals);
        } catch {
          locals = false;
        }

        if (locals) {
          const result = posthtml()
            .use(expressions({locals}))
            .process(source, {sync: true});
          source = result.html;
        }
      //...
```

Comment (Scrum) [reply lib/index.js:26]: we can set the value immediately to default
```js
      // ...
     let locals = false;

      if (src) {
        src = path.resolve(options.root, src);
        source = fs.readFileSync(src, options.encoding);

        try {
          const localsRaw =  node.attrs.locals || node.content ? node.content.join().replace(/\n/g, '') : null;
          locals = JSON.parse(localsRaw);
        } catch {}

        if (locals) {
          const result = posthtml()
            .use(expressions({locals}))
            .process(source, {sync: true});
          source = result.html;
        }
      //...
```

**Review by Scrum — APPROVED** [review]

**Comment by Scrum** [comment]
@felippe-regazio publish v1.5.0

**Pull request merged**